src/movies.js

```diff
@@ -1,16 +1,179 @@
+            return tot
+        }
+    }, 0)
+    if (numOfMovies === 0) {
+        return 0
+    }
+    let avg = avgTot / numOfMovies
+
+    avg = avg.toFixed(2)
+    avg = Number(avg)
+
+    return avg
+}
+
+
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
 
+function orderByYear(movies) {
+    if (movies.length === 0) {
+        return 0
+    }
+    let copyMovie = JSON.parse(JSON.stringify(movies))
+    let sortedMovies = copyMovie.sort((a, b) => {
+        if (a.year > b.year) {
+            return 1
+        } else if (a.year < b.year) {
+            return -1
+        } else {
+            if (a.title > b.title) {
+                return 1
+            } else if (a.title < b.title) {
+                return -1
+            } else {
+
+                return 0
+            }
+            return 0
+        }
+    })
+    return sortedMovies
+}
+
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(movies) {
+    let copyMovies = JSON.parse(JSON.stringify(movies))
+    let ordByAlph = copyMovies.sort((a, b) => {
+        if (a.title > b.title) {
+            return 1
+        } else if (a.title < b.title) {
+            return -1
+        } else {
+
+            return 0
+        }
+    }).map((i) => {
+        return i.title
+    })
+
+    return ordByAlph.splice(0, 20)
+}
+
+
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
 
+function turnHoursToMinutes(movies) {
+    let copyMovies = JSON.parse(JSON.stringify(movies))
+    let hourToMinutes = copyMovies.map((i) => {
+        if (i.duration.includes('h') && i.duration.includes('min') && Number(i.duration[4]) == Number) {
+            let minutes = i.duration.slice(0, 1) * 60 + Number(i.duration.slice(3, 4))
+            return i.duration = minutes
+        } else if (i.duration.includes('h') && i.duration.includes('min')) {
+            let minutes = i.duration.slice(0, 1) * 60 + Number(i.duration.slice(3, 5))
+            return i.duration = minutes
+        } else if (i.duration.includes('h')) {
+            let minutes = i.duration.slice(0, 1) * 60
+            return i.duration = minutes
+        } else if (i.duration.includes('min') && Number(i.duration[1]) == Number) {
+            let minutes = i.duration.slice(0, 3)
+            return i.duration = minutes
+        } else if (i.duration.includes('min')) {
+            let minutes = Number(i.duration.slice(0, 2))
+            return i.duration = minutes
+        } else {
+            return 0
+        }
+    })
+    return copyMovies
+}
+
+
+
+
 // BONUS - Iteration 8: Best yearly rate average - Best yearly rate average
+
+function bestYearAvg(movies) {
+    if (movies.length === 0) {
+        return null
+    }
+    let bestYear = ''
+    let bestRate = 0
+    let yearArr = movies.map((i) => {
+        return i.year   
+    })
+    let rateArr = movies.map((i) =>{
+        return i.rate
+    })
+
```

Comment:
having two different arrays might not be the best approach because the you won't be able to match the data (year-rate), that is why the most common approach here is working around objects that hold both the year and the rate. That way using some methods you can extract the required data.